finetuner/__init__.py

```diff
@@ -6,25 +6,29 @@
 __default_tag_key__ = 'finetuner'
 
 # define the high-level API: fit()
-from typing import Dict, Optional, overload, TYPE_CHECKING, Tuple
+from typing import Callable, Optional, overload, TYPE_CHECKING, Tuple
 
 if TYPE_CHECKING:
-    from .helper import AnyDNN, DocumentArrayLike
+    from .helper import AnyDNN, AnyOptimizer, DocumentSequence
+    from .tuner.base import BaseMiner
     from .tuner.summary import Summary
 
 
 # fit interface generated from Tuner
 @overload
 def fit(
     model: 'AnyDNN',  #: must be an embedding model
-    train_data: 'DocumentArrayLike',
-    eval_data: Optional['DocumentArrayLike'] = None,
+    train_data: 'DocumentSequence',
+    eval_data: Optional['DocumentSequence'] = None,
+    preprocess_fn: Optional[Callable] = None,
+    collate_fn: Optional[Callable] = None,
     epochs: int = 10,
+    miner: Optional['BaseMiner'] = None,
```

Comment:
i see you've created `get_default_miner` below, then we can really hidden this parameter.
Another thing is we parse a loss `SiameseLoss` as default, how can we allow user to change `distance_metric`? such as euclidean?

Reply:
> then we can really hidden this parameter.

what parameter

> how can we allow user to change distance_metric?

User can pass loss as an object, instead of string

Reply:
@bwanglzu miner is now part of the loss

Reply:
i mean do we need to expose `miner` to `fit` function? in the end, miner depends on two things:
1. if user already provides pos/neg.
2. if it is a siamese or triplet.

then in the back we can initialise different miner instances automatically, rather than let user pass a miner class instance to the fit fucntion

Reply:
No, ability for the user to set the miner is necessary. This is because in the future we will have different types of negative miners, which the user would have to explicitly select.